api_tests/scripts/apigee_api.py

```diff
@@ -29,36 +29,74 @@ def _generate_uuid():
+
+    @staticmethod
+    def _verify_status_code(status_code: int or str) -> None:
+        """Verifies the status code provided is a valid status code"""
+        if not type(status_code) == int:
+            try:
+                int(status_code)
+            except ValueError:
+                raise TypeError('Status code must only consist of numbers')
+        else:
+            if len(str(status_code)) != 3:
+                raise TypeError('Status code must be a 3 digit number')
+
     def _get_latest_revision(self) -> str:
         url = f"{APIGEE_API_URL}/apis/{self.proxy}/revisions"
 
-        response = self.get(url, headers=self.headers)
+        response = self.session.get(url, headers=self.headers)
         revisions = response.json()
         return revisions[-1]
 
     def create_debug_session(self):
         url = f"{APIGEE_API_URL}/environments/{APIGEE_ENVIRONMENT}/apis/{self.proxy}/revisions/{self.revision}/" \
               f"debugsessions?session={self.session_name}"
 
-        response = self.post(url, headers=self.headers)
+        response = self.session.post(url, headers=self.headers)
 
-        assert self.check_status_code(response, 201), f"Unable to create apigee debug session {self.session_name}"
+        try:
+            if response.status_code != 201:
+                raise ValueError(f"Unable to create apigee debug session {self.session_name}")
+        except ValueError as ve:
+            print(ve)
 
     def _get_transaction_id(self) -> str:
         url = f"{APIGEE_API_URL}/environments/{APIGEE_ENVIRONMENT}/apis/{self.proxy}/revisions/{self.revision}/" \
               f"debugsessions/{self.session_name}/data"
 
-        response = self.get(url, headers=self.headers)
-        assert self.check_status_code(response, 200), f"Unable to get apigee transaction id for {self.session_name}"
+        response = self.session.get(url, headers=self.headers)
+
+        try:
+            if response.status_code != 200:
+                raise ValueError(f"Unable to get apigee transaction id for {self.session_name}")
+        except ValueError as ve:
+                print(ve)            
+
         return response.text.strip('[]').replace("\"", "").strip().split(', ')[0]
 
     def _get_transaction_data(self) -> dict:
         transaction_id = self._get_transaction_id()
         url = f"{APIGEE_API_URL}/environments/{APIGEE_ENVIRONMENT}/apis/{self.proxy}/revisions/{self.revision}/" \
               f"debugsessions/{self.session_name}/data/{transaction_id}"
 
-        response = self.get(url, headers=self.headers)
-        assert self.check_status_code(response, 200), f"Unable to get apigee transaction {transaction_id}"
+        response = self.session.get(url, headers=self.headers)
+        try:
+            if response.status_code != 200:
+                raise ValueError(f"Unable to get apigee transaction {transaction_id}")
+        except ValueError as ve:
```

Comment:
This `except` clause looks unnecessary. Currently, this whole `try/catch` block only prints message without raising exception to the caller. There is also no need for `print` because the exception contains the message. Same for similar codes below.